packages/@aws-cdk/custom-resources/lib/provider-framework/waiter-state-machine.ts

```diff
@@ -0,0 +1,84 @@
+import { Grant, IGrantable, PolicyStatement, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { IFunction } from '@aws-cdk/aws-lambda';
+import { CfnResource, Construct, Duration, Stack } from '@aws-cdk/core';
+
+export interface StateMachineProps {
+  readonly isCompleteHandler: IFunction;
+
+  readonly timeoutHandler: IFunction;
+
+  readonly interval: Duration;
+
+  readonly maxAttempts: number;
+
+  readonly backoffRate: number;
+}
```

Comment:
Some comments? (I know it's not a public API but still :-))

Reply:
lol. added.